homeassistant/components/smtp/notify.py

```diff
@@ -52,53 +47,43 @@
     DEFAULT_PORT,
     DEFAULT_TIMEOUT,
     DOMAIN,
-    ENCRYPTION_OPTIONS,
 )
 
-PLATFORMS = [Platform.NOTIFY]
+# PLATFORMS = [Platform.NOTIFY]
 
 _LOGGER = logging.getLogger(__name__)
 
-PLATFORM_SCHEMA = PLATFORM_SCHEMA.extend(
-    {
-        vol.Required(CONF_RECIPIENT): vol.All(cv.ensure_list, [vol.Email()]),
-        vol.Required(CONF_SENDER): vol.Email(),
-        vol.Optional(CONF_SERVER, default=DEFAULT_HOST): cv.string,
-        vol.Optional(CONF_PORT, default=DEFAULT_PORT): cv.port,
-        vol.Optional(CONF_TIMEOUT, default=DEFAULT_TIMEOUT): cv.positive_int,
-        vol.Optional(CONF_ENCRYPTION, default=DEFAULT_ENCRYPTION): vol.In(
-            ENCRYPTION_OPTIONS
-        ),
-        vol.Optional(CONF_USERNAME): cv.string,
-        vol.Optional(CONF_PASSWORD): cv.string,
-        vol.Optional(CONF_SENDER_NAME): cv.string,
-        vol.Optional(CONF_DEBUG, default=DEFAULT_DEBUG): cv.boolean,
-        vol.Optional(CONF_VERIFY_SSL, default=True): cv.boolean,
-    }
-)
-
 
 def get_service(
     hass: HomeAssistant,
     config: ConfigType,
     discovery_info: DiscoveryInfoType | None = None,
 ) -> MailNotificationService | None:
     """Get the mail notification service."""
-    setup_reload_service(hass, DOMAIN, PLATFORMS)
+    if discovery_info is None:
+        _LOGGER.warning(
+            "The notify platform setup the smtp integration via configuration.yaml "
+            "is deprecated. You config has been migrated to a config entry and "
+            "Should be removed from your configuration.yaml. "
+            "Canceling setup via configuration.yaml"
+        )
+        return None
+    entry_id = discovery_info["entry_id"]
     mail_service = MailNotificationService(
-        config[CONF_SERVER],
-        config[CONF_PORT],
-        config[CONF_TIMEOUT],
-        config[CONF_SENDER],
-        config[CONF_ENCRYPTION],
-        config.get(CONF_USERNAME),
-        config.get(CONF_PASSWORD),
-        config[CONF_RECIPIENT],
-        config.get(CONF_SENDER_NAME),
-        config[CONF_DEBUG],
-        config[CONF_VERIFY_SSL],
+        discovery_info.get(CONF_SERVER, DEFAULT_HOST),
+        discovery_info.get(CONF_PORT, DEFAULT_PORT),
+        discovery_info.get(CONF_TIMEOUT, DEFAULT_TIMEOUT),
+        discovery_info[CONF_SENDER],
+        discovery_info.get(CONF_ENCRYPTION, DEFAULT_ENCRYPTION),
+        discovery_info.get(CONF_USERNAME),
+        discovery_info.get(CONF_PASSWORD),
+        discovery_info[CONF_RECIPIENT],
+        discovery_info.get(CONF_SENDER_NAME),
+        discovery_info.get(CONF_DEBUG, DEFAULT_DEBUG),
+        discovery_info.get(CONF_VERIFY_SSL, True),
     )
 
+    hass.data.setdefault(DOMAIN, {})[entry_id] = mail_service
```

Comment:
```suggestion
    hass.data[DOMAIN][entry_id] = mail_service
```

You're already doing `hass.data[DOMAIN][config_entry.entry_id] = None` in `async_setup_entry`